stack-graphs/src/stats.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::HashMap;
+use std::hash::Hash;
+
+use itertools::Itertools;
+
+/// Frequency distribution maintains the frequency of T values.
+#[derive(Clone, Debug, Default)]
+pub struct FrequencyDistribution<T>
```

Comment:
I'm not sure if I consider it in scope for this PR, but we might consider a hyperloglog crate of some kind instead of rolling our own histogram type.